### PR TITLE
Feat: designated deposits withdraw

### PR DESCRIPTION
**Pull request by mrLSD**
## Summary by CodeRabbit

* **New Features**
  * Configurable deposit distribution between solver and fee accounts with percentage validation.
  * On-chain action to distribute deposit tokens (single- and multi-token), status checks, and tracking of distributed totals to block premature admin withdrawals.
  * Admin withdrawal guarded until deposit distribution is complete.

* **Refactor**
  * Renamed sale token distribution action to a sale-specific name.

* **Tests**
  * Added extensive integration tests covering deposit distribution, edge cases, refunds, and admin withdrawal flows.

### CODE REVIEW

File: contract/src/admin_withdraw.rs [hunk 1]
```diff
@@ -1,23 +1,48 @@
...
 const GAS_FOR_MT_BALANCE_OF: Gas = Gas::from_tgas(1);
 const GAS_FOR_MT_TRANSFER: Gas = Gas::from_tgas(5);
 const GAS_WITHDRAW_NEP141_CALLBACK: Gas = Gas::from_tgas(50);
 const GAS_WITHDRAW_NEP245_CALLBACK: Gas = Gas::from_tgas(60);
+const GAS_FOR_FINISH_ADMIN_WITHDRAW: Gas = Gas::from_tgas(5);
 
 #[near]
 impl AuroraLaunchpadContract {
+    #[payable]
+    pub fn withdraw_deposits(&mut self, direction: AdminWithdrawDirection) -> Promise {
```
Comment (birchmd) [inline contract/src/admin_withdraw.rs:22]: +1 for coderabbit pointing out the role should be restricted.

But I wonder if we need this function at all. It looks like identical behaviour would happen by calling `admin_withdraw(WithdrawalToken::Deposit, direction, None)`. Do we need a function that is only a special case of another function?

**Review by aleksuss — APPROVED** [review]

File: contract/src/distribute/deposit.rs [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+        }
+    }
+
+    fn distribute_nep141_deposit_tokens(
+        &self,
+        token_account_id: &AccountId,
+        solver_amount: u128,
+        fee_amount: u128,
+    ) -> Promise {
+        let solver_promise = (solver_amount > 0).then_some(
```
Comment (birchmd) [inline contract/src/distribute/deposit.rs:58]:
```suggestion
        let solver_promise = (solver_amount > 0).then(
```

I think it makes sense to use `then` instead of `then_some` because the argument to the latter is always evaluated whereas the former takes a closure that is only evaluated when the bool is `true`.

This comment applies to all `then_some` calls in this file.
